src/main/java/com/telefonica/iot/tidoop/backends/ckan/CKANBackend.java

```diff
@@ -198,6 +210,110 @@ public JSONArray getRecords(String resId, long start, long end) {
+        try {
+            // create the CKAN request JSON
+            String jsonString = "{ \"name\": \"" + resName + "\", "
+                    + "\"url\": \"none\", "
+                    + "\"package_id\": \"" + pkgId + "\" }";
+            
+            // create the CKAN request URL
+            String urlPath = "http" + (ssl ? "s" : "") + "://" + ckanHost + ":" + ckanPort
+                    + "/api/3/action/resource_create";
+            
+            // do the CKAN request
+            CKANResponse res = doCKANRequest("POST", urlPath, jsonString);
+
+            // check the status
+            if (res.getStatusCode() == 200) {
+                String resId = ((JSONObject) res.getJsonObject().get("result")).get("id").toString();
+                logger.debug("Successful resource creation (resName/resId=" + resName + "/" + resId + ")");
+                return resId;
+            } else {
+                throw new Exception("Don't know how to treat the response code. Possibly the resource "
+                        + "already exists (respCode=" + res.getStatusCode() + ", resourceName=" + resName + ")");
+            } // if else if else
+        } catch (Exception e) {
+            throw new Exception(e.getMessage());
+        } // try catch
+    } // createResource
+    
+    /**
+     * Creates a datastore for a given resource in CKAN.
+     * @param resId Identifies the resource whose datastore is going to be created
+     * @throws Exception
+     */
+    public void createKeyValueDatastore(String resId) throws Exception {
+        try {
+            // create the CKAN request JSON
+            String jsonString = "{ \"resource_id\": \"" + resId
+                    + "\", \"fields\": [ "
+                    + "{ \"id\": \"key\", \"type\": \"text\"},"
+                    + "{ \"id\": \"value\", \"type\": \"int\"}"
+                    + "], "
+                    + "\"force\": \"true\" }";
+            
+            // create the CKAN request URL
+            String urlPath = "http" + (ssl ? "s" : "") + "://" + ckanHost + ":" + ckanPort
+                    + "/api/3/action/datastore_create";
+            
+            // do the CKAN request
+            CKANResponse res = doCKANRequest("POST", urlPath, jsonString);
+
+            // check the status
+            if (res.getStatusCode() == 200) {
+                logger.debug("Successful datastore creation (resourceId=" + resId + ")");
+            } else {
+                throw new Exception("Don't know how to treat the response code. Possibly the datastore "
+                        + "already exists (respCode=" + res.getStatusCode() + ", resourceId=" + resId + ")");
+            } // if else if else
+        } catch (Exception e) {
+            throw new Exception(e.getMessage());
+        } // try catch
+    } // createKeyValueDatastore
+    
+    /**
+     * Inserts a (key, value) pair into a givern resource.
```

Comment:
Fixed in https://github.com/telefonicaid/fiware-tidoop/commit/43a8b52155e83e6cc2096b18679049cf607f2a62